dqvm/src/transition.lisp

```diff
@@ -0,0 +1,68 @@
+;;;; src/transition.lisp
+;;;;
+;;;; Author: Juan M. Bello-Rivas
+;;;;         Robert Smith
+
+(in-package #:dqvm2)
+
+(defmethod transition :around (qvm instr)
+  ;; Ideally, we should make *trace-output* be a gateway to cl-syslog instead of adapting QVM's code here.
+  (cond
+    ((not qvm:*transition-verbose*) (call-next-method))
+    (t
+     (let ((start (get-internal-real-time))
+           gc-time bytes-alloc)
+       (multiple-value-prog1 (qvm::measuring-gc (gc-time bytes-alloc) (call-next-method))
+         ;; (format-log :debug "~A" qvm)
+
+         (format-log :debug "Transition ~A took ~D ms (gc: ~D ms; alloc: ~D bytes)"
+                     (with-output-to-string (s) (cl-quil:print-instruction instr s))
```

Comment:
Would `~/cl-quil:instruction-fmt/` work here?

Reply:
Definitely! Thanks for the tip.